### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by cs-utkarsh-surana**
### Descriptions: 
- Added the option to configure time to change the message.
- Added the option to change the value of picklist of the view pannel.

#### UTCs:  

- [x] Widget installation verified.  
- [x] Tested the edit page.
- [x] Tested the view page.

### CODE REVIEW

File: widget/edit.controller.js [hunk 1]
```diff
@@ -58,6 +71,21 @@
...
          });
       };
+
+      function loadAttributestoupdate() {
+        $scope.newfieldsArray = [];
+        var entity = new Entity($state.params.module);
+        entity.loadFields().then(function () {
+          $scope.newfieldsArray = entity.getFormFieldsArray();
+          $scope.newfieldsArray = _.filter($scope.newfieldsArray, function(item){ return item.model === 'picklists'; });
+          if(config.newpicklistItem){
```
Comment (cs-pranali-chavan) [inline widget/edit.controller.js]: $scope.config.newpicklistItem will be used. please verify

File: widget/edit.controller.js [hunk 2]
```diff
@@ -58,6 +71,21 @@
...
+            var selectedPicklistItem = _.find($scope.newfieldsArray, function(item){
+            return item.name === config.newpicklistItem;
```
Comment (cs-pranali-chavan) [inline widget/edit.controller.js]: $scope.config.newpicklistItem will be used. please verify here as well

File: widget/view.controller.js [hunk 3]
```diff
@@ -1,72 +1,121 @@
...
+       var widgetsubscription;
+ 
+       function checkPicklistCondition() {
+          Modules.get({
+             module: $state.params.module,
+             id: $state.params.id,
+             __selectFields: config.picklistItem
+          }).$promise.then(function (result) {
+             var foundPickListValue = _.find($scope.config.picklits_mapping.options, function (item) {
+                return item.picklistValue === result[config.picklistItem].itemValue;
```
Comment (cs-pranali-chavan) [inline widget/view.controller.js]: $scope.config.picklistItem will be used. please verify

File: widget/view.controller.js [hunk 4]
```diff
@@ -1,72 +1,121 @@
...
+                $scope.showSpinner = false;
+                $scope.showIcon = false;
+                $scope.loading = false;
+             }
+          })
+       }
+ 
+        function updatePicklist() {
+          var picklistObject = {};
+          picklistObject[config.newpicklistItem] = $scope.selectedPicklistObject
```
Comment (cs-pranali-chavan) [inline widget/view.controller.js]: $scope.config.picklistItem will be used. please verify

**Review by cs-pranali-chavan — CHANGES_REQUESTED** [review]
Its better to use the $scope attribute once the parameter is assigned. Rest changes look good.

File: widget/edit.controller.js (hunk 2 above)
Comment (cs-utkarsh-surana) [reply widget/edit.controller.js]: updated

File: widget/view.controller.js (hunk 3 above)
Comment (cs-utkarsh-surana) [reply widget/view.controller.js]: updated

File: widget/view.controller.js (hunk 4 above)
Comment (cs-utkarsh-surana) [reply widget/view.controller.js]: updated

**Pull request merged**